File: e036.js
/**Checks if a number is palindrome */
function ispal(n) {
    let a = Array.from(n.toString());
    let b = a.map(i => i).reverse();
    return a.toString() === b.toString();
}

let sum = 0;
for (let i = 1; i < 1000000; i = i + 2) {
    if (ispal(i) && ispal(i.toString(2))) sum += i;
}
console.log(sum);

File: e036.py
"""
Problem 36: Double-base palindrome
"""
from time import time


def ispal(n):
    """Checks if a number is palindrome"""
    return str(n) == str(n)[::-1]


def bin2(num):
    """Converts a number in base10 to base2"""
    return int(bin(num)[2:])


tic = time()
print(sum(i for i in range(1, 1000000, 2) if ispal(i) and ispal(bin2(i))))
print(time() - tic)
